### ISSUE TITLE
lint_package() and find_package(".") does not work anymore if depth >= 2

### ISSUE THREAD

**Issue by BenoitLondon** [r-lib/lintr#1759]
In my package I have a code style test so it calls `lint_package()` from the `tests/testthat` folder.
The new logic in find_package fails at finding the root of the package ( I guess from this commit https://github.com/r-lib/lintr/commit/3009505944493acdaeeec6ff819acede1f39250a )

so If I call lint_package from a deeper level than 2 it fails at finding the package actually...
```
test_that("check that package has code style", {
        skip_on_covr()
        lints <- lintr::lint_package()
        ....
        }
```
The above code warns/fails with:

```
Warning message:
In lintr::lint_package() :
  Didn't find any R package searching upwards from '.'.
  ```

**Comment by AshesITR** [comment]
Hi, can you give us an MWE so we can see what is going on?

**Comment by AshesITR** [comment]
Thanks for providing the MWE.
I can confirm this, the problem is actually not related to the nesting (2 is allowed):

```
str(lintr::lint_package("~/testme/dir1/dir2"))

 Named list()
 - attr(*, "class")= chr "lints"
 - attr(*, "path")= chr "/home/alex/testme"
```

So the problem is actually that the path needs to be normalized before starting the search.

**Comment by IndrajeetPatil** [comment]
With #1765, no more warnings: 

``` r
usethis::create_package("~/testme", open = FALSE)
#> ✔ Setting active project to '/Users/indrajeetpatil/testme'
#> Package: testme
#> Title: What the Package Does (One Line, Title Case)
#> Version: 0.0.0.9000
#> Authors@R (parsed):
#>     * First Last <first.last@example.com> [aut, cre] (YOUR-ORCID-ID)
#> Description: What the package does (one paragraph).
#> License: `use_mit_license()`, `use_gpl3_license()` or friends to pick a
#>     license
#> Encoding: UTF-8
#> Roxygen: list(markdown = TRUE)
#> RoxygenNote: 7.2.1
#> ✔ Setting active project to '<no active project>'
dir.create("~/testme/dir1/dir2", recursive = TRUE)
#> Warning in dir.create("~/testme/dir1/dir2", recursive = TRUE): '/Users/
#> indrajeetpatil/testme/dir1/dir2' already exists
setwd("~/testme/dir1/dir2")
lintr::lint_package()
```

<sup>Created on 2022-11-09 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by BenoitLondon** [comment]
thanks for the fix!
